src/autogluon/cloud/backend/sagemaker_backend.py

```diff
@@ -472,13 +472,16 @@ def deploy(
         if deploy_kwargs is None:
             deploy_kwargs = {}
 
+        # VolumeSize parameter is not allowed for the selected Instance type ml.g4dn.* or ml.p3.*
+        if not instance_type.startswith(tuple(["ml.g", "ml.p"])):
+            deploy_kwargs.update(volume_size=volume_size)
```

Comment:
Let's add a warning to let the user know that the volume size is not taking effect